src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1857,6 +1862,24 @@ MavlinkReceiver::handle_message_obstacle_distance(mavlink_message_t *msg)
 	_obstacle_distance_pub.publish(obstacle_distance);
 }
 
+void
+MavlinkReceiver::handle_message_tunnel(mavlink_message_t *msg)
+{
+	mavlink_tunnel_t mavlink_tunnel;
+	mavlink_msg_tunnel_decode(msg, &mavlink_tunnel);
+
+	tunnel_s tunnel{};
+
+	tunnel.timestamp = hrt_absolute_time();
+	tunnel.payload_type = mavlink_tunnel.payload_type;
+	tunnel.target_system = mavlink_tunnel.target_system;
+	tunnel.target_component = mavlink_tunnel.target_component;
+	tunnel.payload_length = mavlink_tunnel.payload_length;
+	memcpy(tunnel.payload, mavlink_tunnel.payload, sizeof(tunnel.payload));
```

Comment:
Probably just being paranoid, but you could add a static assert that the payloads are the same size. The mavlink message is technically work in progress and can change.

Reply:
Yes, this is a better way. Thanks.